src/server/auth.ts

```diff
@@ -107,6 +108,37 @@ export const authOptions: NextAuthOptions = {
 				});
 			},
 		},
+		// {
+		// 	id: 'azure-ad',
+		// 	name: 'Azure Active Directory',
+		// 	type: 'oauth',
+		// 	version: '2.0',
+		// 	authorization: { params: { scope: "openid profile user.Read email" } },
+		// 	accessTokenUrl: 'https://login.microsoftonline.com/{tenantID}/oauth2/v2.0/token',
+		// 	authorizationUrl: 'https://login.microsoftonline.com/{tenantID}/oauth2/v2.0/authorize',
+		// 	clientId: process.env.AZURE_AD_CLIENT_ID,
+		// 	clientSecret: process.env.AZURE_AD_CLIENT_SECRET,
+		// },		
+		AzureADProvider({	
+			// id: "azure-ad",
+			// name: "Active Directory Azure AD ",
+			// checks: ["state", "pkce"],
+			// authorization: { params: { scope: "openid profile user.Read email" } },
+			allowDangerousEmailAccountLinking:
+				Boolean(process.env.ALLOW_DANGEROUS_EMAIL_LINKING) || true,
+			clientId: process.env.AZURE_AD_CLIENT_ID,
+			clientSecret: process.env.AZURE_AD_CLIENT_SECRET,
+			tenantId: process.env.AZURE_AD_TENANT_ID,
+			profile(profile) {
+				return Promise.resolve({
+					id: profile.sub || profile.id.toString(), // Handle ID based on provider
+					name: profile.name || profile.login || profile.username,
+					email: profile.email,
+					image: profile.picture || profile.avatar_url || profile.image_url || profile.image,
+					role: "USER",
+				});
+			},
+		}),
 		CredentialsProvider({
```

Comment:
remove all unused code.
also, i would prefer if we could incorporate this into the custom oauth adapter. I dont want to add code for all providers out there.